src/commands/usages/index.ts

```diff
@@ -0,0 +1,181 @@
+            const includeGlobs = flags['include'] || codeInsightsConfig.includeFiles
+            return includeGlobs
+                ? includeGlobs.some((glob) => minimatch(filepath, glob, { matchBase: true }))
+                : true
+        }
+
+        const excludeFile = (filepath: string) => {
+            const excludeGlobs = flags['exclude'] || codeInsightsConfig.excludeFiles
+            return excludeGlobs
+                ? excludeGlobs.some((glob) => minimatch(filepath, glob, { matchBase: true }))
+                : false
+        }
+
+        const processFile = (filepath: string): File => {
+            let lines: LineItem[] = []
+            try {
+                lines = fs
+                    .readFileSync(filepath, 'utf8')
+                    .split('\n')
+                    .map((content, idx) => ({ content, ln: idx + 1 }))
+            } catch (err) {
+                this.warn(`Error parsing file ${filepath}`)
+                this.debug(err)
+            }
+            return {
+                name: filepath,
+                lines
+            }
+        } 
+
+        const files = lsFiles()
+            .filter((filepath) => includeFile(filepath) && !excludeFile(filepath))
+            .map(processFile)
+
+        if (!files.length) {
+            this.warn('No files found to process.')
+            return
+        }
+
+        const matchesBySdk = parseFiles(files, {
+            clientNames: getClientNames(flags, this.configFromFile),
+            matchPatterns: getMatchPatterns(flags, this.configFromFile),
+            printPatterns: showRegex(flags)
+        })
+
+        const variableAliases = getVariableAliases(flags, this.configFromFile)
+        
+        const matchesByVariable = this.getMatchesByVariable(matchesBySdk, variableAliases)
+        
+        if (flags['format'] === 'json') {
+            const matchesByVariableJSON = this.formatMatchesToJSON(matchesByVariable)
+            this.log(JSON.stringify(matchesByVariableJSON))
+        } else {
+            this.formatConsoleOutput(matchesByVariable)
+        }
+    }
+
+    private getMatchesByVariable(
+        matchesBySdk: Record<string, VariableUsageMatch[]>,
+        aliasMap: Record<string, string>
+    ) {
+        const matchesByVariable: Record<string, VariableUsageMatch[]> = {}
+        Object.values(matchesBySdk).forEach((matches) => {
+            matches.forEach((m) => {
+                const match = { ...m }
+                const aliasedName = aliasMap[match.name]
+                if (match.isUnknown && aliasedName) {
+                    match.alias = match.name
+                    match.name = aliasedName
+                    delete match.isUnknown
+                }
+                if (!match.isUnknown) {
+                    matchesByVariable[match.name] ??= []
+                    matchesByVariable[match.name].push(match)
+                    matchesByVariable[match.name] = uniqBy(
```

Comment:
minor thing but it seems inefficient to be redoing the uniqBy on each iteration. Maybe you could use a set here? Or do the uniqBy at the end for each array in the object?